src/providers/oracle/qgsoracleprovider.h

```diff
@@ -157,7 +157,7 @@ class QgsOracleProvider : public QgsVectorDataProvider
     QString description() const override;
     QgsFeatureIterator getFeatures( const QgsFeatureRequest &request = QgsFeatureRequest() ) const override;
 
-    static bool exec( QSqlQuery &qry, QString sql, const QVariantList &args );
+    static bool exec( QSqlQuery &qry, QString sql, const QVariantList &args, bool retry = false );
```

Comment:
Just an idea, how about making this a `int retryCount = 0` and within the recursion do ` return exec( ... , retryCount -1 )`?